r/R/expression.R

```diff
@@ -181,6 +181,20 @@ find_array_refs <- function(x) {
   unlist(out)
 }
 
+get_field_name <-  function(x, msg = NULL) {
```

Comment:
How does this relate to `get_field_names()` in dplyr.R?

Reply:
I should probably rename this. It takes a single `array_expression` as input. If the expression is just a bare column reference, it returns its name deparsed as a character string. If the expression is something more than just a bare column reference, it errors. This function will change or go away when I implement support for sorting by expressions.

Reply:
This basically looks like the anonymous function inside of `map_chr` in `get_field_names()`. Whether or not you end up deciding that you need it here, you could factor that out as `get_field_name()` so it is usable in cases like this. (It also works on dataset Expressions, so you'll be set for supporting this feature on Datasets too.)

Reply:
This is moot after 0ba20fb57c542e3099b047c792579819685c8888 which removes this function